### PR TITLE
FEATURE: Allow voting on the first post of a topic.

### PR DESCRIPTION
**Pull request by tgxworld**
Discussed this internally and we decided that the first post of a topic should be votable. This allows us to sort questions in the future by votes. 

### Screenshot

Screenshot from 2022-02-08 13-54-21

### CODE REVIEW

File: app/models/question_answer_vote.rb [hunk 1]
```diff
@@ -31,7 +31,7 @@ def self.reverse_direction(direction)
   def ensure_valid_post
     if !post.qa_enabled
       errors.add(:base, I18n.t("post.qa.errors.qa_not_enabled"))
-    elsif post.post_number == 1 || post.reply_to_post_number.present?
+    elsif post.reply_to_post_number.present?
```
Comment (nattsw) [inline app/models/question_answer_vote.rb:34]: Thinking out loud: `post.reply_to_post_number.present?` means "this post is an answer"?

**Review by nattsw — APPROVED** [review]

**Comment by nattsw** [comment]
Btw thanks for the screenshots, they're great!!

Also because of that, I'm wondering if you'd be interested in making the comments have less whitespace, similar to SO.

> this is my comment - nat

File: app/models/question_answer_vote.rb (hunk 1 above)
Comment (tgxworld) [reply app/models/question_answer_vote.rb:34]: A post can be in reply to another post and we use this column in core to indicate that.

**Comment by tgxworld** [comment]
Thank you for the review @nattsw :+1:

**Pull request merged**